src/message.h

```diff
@@ -30,7 +30,7 @@ namespace rhost {
 
         struct message_repr {
             boost::endian::little_uint64_buf_t id, request_id;
-            char data[];
```

Comment:
(The reason for the warning is that this is technically a non-standard C++ extension. However, all C++ compilers support it in practice.)